mev_inspect/block.py

```diff
@@ -36,43 +39,64 @@ async def create_from_block_number(
-        block = await _fetch_block(w3, base_provider, block_number)
-        return block
-    else:
+        block = await _fetch_block(w3, base_provider, geth, block_number)
         return block
+    return block
 
 
-async def _fetch_block(w3, base_provider, block_number: int, retries: int = 0) -> Block:
-    block_json, receipts_json, traces_json, base_fee_per_gas = await asyncio.gather(
-        w3.eth.get_block(block_number),
-        base_provider.make_request("eth_getBlockReceipts", [block_number]),
-        base_provider.make_request("trace_block", [block_number]),
-        fetch_base_fee_per_gas(w3, block_number),
-    )
-
-    try:
-        receipts: List[Receipt] = [
-            Receipt(**receipt) for receipt in receipts_json["result"]
-        ]
-        traces = [Trace(**trace_json) for trace_json in traces_json["result"]]
-    except KeyError as e:
-        logger.warning(
-            f"Failed to create objects from block: {block_number}: {e}, retrying: {retries + 1} / 3"
+async def _fetch_block(
+    w3, base_provider, geth: bool, block_number: int, retries: int = 0
+) -> Block:
+    if not geth:
+        block_json, receipts_json, traces_json, base_fee_per_gas = await asyncio.gather(
+            w3.eth.get_block(block_number),
+            base_provider.make_request("eth_getBlockReceipts", [block_number]),
+            base_provider.make_request("trace_block", [block_number]),
+            fetch_base_fee_per_gas(w3, block_number),
         )
-        if retries < 3:
-            await asyncio.sleep(5)
-            return await _fetch_block(w3, base_provider, block_number, retries)
-        else:
-            raise
 
-    return Block(
-        block_number=block_number,
-        block_timestamp=block_json["timestamp"],
-        miner=block_json["miner"],
-        base_fee_per_gas=base_fee_per_gas,
-        traces=traces,
-        receipts=receipts,
-    )
+        try:
+            receipts: List[Receipt] = [
+                Receipt(**receipt) for receipt in receipts_json["result"]
+            ]
+            traces = [Trace(**trace_json) for trace_json in traces_json["result"]]
+            return Block(
+                block_number=block_number,
+                block_timestamp=block_json["timestamp"],
+                miner=block_json["miner"],
+                base_fee_per_gas=base_fee_per_gas,
+                traces=traces,
+                receipts=receipts,
+            )
+        except KeyError as e:
+            logger.warning(
+                f"Failed to create objects from block: {block_number}: {e}, retrying: {retries + 1} / 3"
+            )
+            if retries < 3:
+                await asyncio.sleep(5)
+                return await _fetch_block(
+                    w3, base_provider, geth, block_number, retries
+                )
+            else:
+                raise
+    else:
+        # print(block_number)
+        block_json = await asyncio.gather(w3.eth.get_block(block_number))
```

Comment:
can remove gather here since it's just one entry and await instead

Reply:
Doesn't seem to work without asyncio.gather